packages/next-codemod/bin/upgrade.ts

```diff
@@ -74,14 +74,31 @@ export async function runUpgrade(
     return
   }
 
+  // If the app is a pure Pages Router (no app dir) and the user wants to stay on React 18,
+  // we can skip the upgrading React and suggesting relevant codemods.
+  const { isPagesOnlyAndWantReact18 } = await prompts(
```

Comment:
Let's only ask that if you're above .45 and still have 18 installed. Saves one prompt for the good citizen who are on 19.

Technically you can't withhold if you're between .45 and whatever version we landed support for React 18 in pages router (.169?) but that's not really a version range we need to be concerned with.